### DISCUSSION TITLE
how to handle deployments while node-maintenance

### DISCUSSION THREAD

**Discussion by staabm** [deployphp/deployer#2394] [Help needed]
we are deploying to 2 nodes in parallel. from time to time a single node might go into maintenance mode, get restarted or go into similar non-available states and therefore is not connectable at deployment time.

is there a recommendation for this kind of scenario? 

in our case the deployment fails because of node2 is not available, even when node1 is actually available and serves regular traffic.
I am not sure if there is a clever way on how to handle this kind of situation

**Comment by antonmedv** [comment]
What I would do is create a shell script which will trigger deploys to two hosts in parallel. If one fails, other will succeed.

**Reply by staabm** [reply]
Thx for the suggestion. 

Do you also have an idea how to make sure that servers comming back from maintenance can make sure they contain the latest deployment before getting added back to the cluster?

**Reply by antonmedv** [reply]
In v7 I create a command `dep status` it shows currently deployed info. Maybe it can help.
